icgcget/commands/download.py

```diff
@@ -0,0 +1,204 @@
+        return object_ids
+
+    def download(self, object_ids, staging, output,
+                 cghub_access, cghub_path, cghub_transport_parallel,
+                 ega_access, ega_path, ega_transport_parallel, ega_udt,
+                 gdc_access, gdc_path, gdc_transport_parallel, gdc_udt,
+                 icgc_access, icgc_path, icgc_transport_file_from, icgc_transport_parallel,
+                 pdc_access, pdc_path, pdc_transport_parallel):
+
+        if 'cghub' in object_ids and object_ids['cghub']:
+            check_access(self, cghub_access, 'cghub')
+            self.gt_client.session = object_ids
+            uuids = self.get_uuids(object_ids['cghub'])
+            return_code = self.gt_client.download(uuids, cghub_access, cghub_path, staging, cghub_transport_parallel)
+            object_ids = self.icgc_client.session
+            self.check_code('Cghub', return_code)
+            self.move_files(staging, output)
+
+        if 'aws-virginia' in object_ids and object_ids['aws-virginia']:
+            check_access(self, icgc_access, 'icgc')
+            self.icgc_client.session = object_ids
+            uuids = self.get_uuids(object_ids['aws-virginia'])
+            return_code = self.icgc_client.download(uuids, icgc_access, icgc_path, staging, icgc_transport_parallel,
+                                                    file_from=icgc_transport_file_from, repo='aws')
+            object_ids = self.icgc_client.session
+            self.check_code('Icgc', return_code)
+            self.move_files(staging, output)
+
+        if 'ega' in object_ids and object_ids['ega']:
+            check_access(self, ega_access, 'ega')
+            if ega_transport_parallel != '1':
+                self.logger.warning("Parallel streams on the ega client may cause reliability issues and failed " +
+                                    "downloads.  This option is not recommended.")
+            self.ega_client.session = object_ids
+            uuids = self.get_uuids(object_ids['ega'])
+            return_code = self.ega_client.download(uuids, ega_access, ega_path, staging, ega_transport_parallel, ega_udt)
+            object_ids = self.icgc_client.session
+            self.check_code('Ega', return_code)
+            self.move_files(staging, output)
+
+        if 'collaboratory' in object_ids and object_ids['collaboratory']:
+            check_access(self, icgc_access, 'icgc')
+            self.icgc_client.session = object_ids
+            uuids = self.get_uuids(object_ids['collaboratory'])
+            return_code = self.icgc_client.download(uuids, icgc_access, icgc_path, staging, icgc_transport_parallel,
+                                                    file_from=icgc_transport_file_from, repo='collab')
+            object_ids = self.icgc_client.session
+            self.check_code('Icgc', return_code)
+            self.move_files(staging, output)
+
+        if 'gdc' in object_ids and object_ids['gdc']:
```

Comment:
You commented on this three hours ago.  I did fix it, but haven't committed recently, because the code hasn't been in a working state.